src/core/expr/fexpr_count_countna.cc

```diff
@@ -0,0 +1,165 @@
+//------------------------------------------------------------------------------
+// Copyright 2023 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "column/const.h"
+#include "column/latent.h"
+#include "column/countna.h"
+#include "column/count_all_rows.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+namespace dt {
+namespace expr {
+
+template<bool COUNTNA>
+class FExpr_CountNA : public FExpr_Func {
+  private:
+    ptrExpr arg_;
+
+  public:
+    FExpr_CountNA(ptrExpr &&arg)
+      : arg_(std::move(arg)) {}
+
+    std::string repr() const override {
+      std::string out = COUNTNA? "countna" : "count";
+      out += '(';
+      if (arg_->get_expr_kind() != Kind::None) out += arg_->repr();      
+      out += ')';
+      return out;
+    }
+
+
+    Workframe evaluate_n(EvalContext &ctx) const override {
+      Workframe outputs(ctx);
+      Workframe wf = arg_->evaluate_n(ctx);
+      Groupby gby = ctx.get_groupby();
+      // this covers scenarios where
+      // we dont care about the presence or absence of NAs
+      // we just want the total number of rows
+      bool count_all_rows = arg_->get_expr_kind() == Kind::None;
+
+      if (count_all_rows && !COUNTNA) {
+        Column coli;
+        if (gby){
+          coli = Column(new Latent_ColumnImpl(
+                new CountAllRows_ColumnImpl(gby)
+              ));
+        } else {
+            auto value = static_cast<int64_t>(ctx.nrows());
+            coli = Const_ColumnImpl::make_int_column(1, value, SType::INT64);
+          }
```

Comment:
Same here.

Reply:
what am I missing with the indetation here? the `else` curly brackets line up

Reply:
The code for both if and else should have the same indentation. In this case the code that goes after else should be moved two spaces to the left, including the closing bracket.